### ISSUE TITLE
sft-loss NaN

### ISSUE THREAD

**Issue by yiyepiaoling0715** [volcengine/verl#467]
the top N steps is normal,loss is close 2 0,and then suddenly become NaN,what reason maybe?
`torchrun --standalone --nnodes=${NODE_NUM} --nproc_per_node=${GPU_NUM} \
     -m verl.trainer.fsdp_sft_trainer \
    data.train_files=$train_files \
    data.val_files=$test_files \
    data.prompt_key=prompt \
    data.response_key=answer \
    data.train_batch_size=16 \
    data.micro_batch_size_per_gpu=2 \
    data.max_length=4096 \
    model.partial_pretrain=$model_path \
    model.enable_gradient_checkpointing=true \
    model.trust_remote_code=true \
    trainer.default_local_dir=${LPAI_CHECKPOINT} \
    trainer.project_name=fullfile-sft \
    trainer.experiment_name=fullfile-deepseek-coder-6.7b-instruct \
    trainer.total_epochs=${NUM_TRAIN_EPOCHS} \
    trainer.logger=['console','wandb','tensorboard'] \
    trainer.default_hdfs_dir=null \
    optim.lr=1e-5 $@

`
![Image](https://github.com/user-attachments/assets/b9f91dc1-e071-43ed-b7ec-c03123ce1e80)

**Comment by Kirrito-k423** [comment]
中文：感谢反馈。仅凭当前 issue 中的命令和 loss 曲线，还不足以判断这是 verl 的确定代码缺陷。SFT loss 的计算依赖具体 batch 中的 loss_mask/response_mask、有效 token 数、模型 log_probs、精度/优化器状态以及数据格式；目前缺少 verl 版本或 commit、可复现数据样例、首次 NaN 前后的完整日志、grad_norm/lr、硬件与混合精度配置。另一个近期 timeline reference 指向外部 fork 的依赖调整 commit，和该 SFT NaN 问题没有直接解决关系。

建议补充：1) 使用的 verl commit/tag；2) 一小段可复现 parquet/json 数据或脱敏样例及 schema；3) 完整展开后的训练命令和配置；4) 首次 NaN 前后 20-50 step 的 console/wandb 日志，包括 loss、grad_norm、lr、显存/精度信息；5) GPU 型号、torch/transformers/flash-attn 版本。拿到这些信息后才能进一步判断是数据掩码、数值稳定性、学习率/精度，还是 SFT trainer 的问题。

English: Thanks for reporting this. The current issue only includes the training command and a loss curve description, which is not enough to identify a definite verl bug. SFT loss depends on the batch loss_mask/response_mask, valid token count, model log_probs, precision/optimizer state, and dataset format. The report is missing the verl version or commit, a minimal reproducible data sample, complete logs around the first NaN, grad_norm/lr, hardware details, and mixed precision settings. The recent timeline reference points to an external fork commit that only changes an OpenAI dependency constraint, so it does not appear to resolve this SFT NaN issue.

Please provide: 1) the verl commit/tag; 2) a small reproducible parquet/json sample or sanitized schema; 3) the fully expanded training command/config; 4) 20-50 steps of logs around the first NaN including loss, grad_norm, lr, memory/precision information; 5) GPU model and torch/transformers/flash-attn versions. With that, maintainers can check whether this is caused by data masking, numerical instability, learning rate/precision settings, or the SFT trainer implementation.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
